homeassistant/components/mqtt/__init__.py

```diff
@@ -754,9 +754,10 @@ class Subscription:
 class MqttClientSetup:
     """Helper class to setup the paho mqtt client from config."""
 
-    # We don't import on the top because some integrations
-    # should be able to optionally rely on MQTT.
-    import paho.mqtt.client as mqtt  # pylint: disable=import-outside-toplevel
+    if TYPE_CHECKING:
+        # Only import for type checking here, imports are done locally
+        # because integrations should be able to optionally rely on MQTT.
+        import paho.mqtt.client as mqtt  # pylint: disable=import-outside-toplevel
```

Comment:
Isn't this a class attribute? How can it be only for type checking? Where is the mqtt attribute set otherwise?

Reply:
It's needed for type checking of:
```py
def client(self) -> mqtt.Client:
```

Reply:
We could of course move the conditional import it at top level instead if you prefer that?

Reply:
As I said, it looks like this is a class attribute and not only needed for type checking.

Reply:
You're of course right, i did not notice the import was used as a class attribute.